cinema/views.py

```diff
@@ -1,45 +1,99 @@
+    CinemaHallSerializer,
+    GenreSerializer
+)
 
 
-@api_view(["GET", "POST"])
-def movie_list(request):
-    if request.method == "GET":
-        movies = Movie.objects.all()
-        serializer = MovieSerializer(movies, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+class GenreList(APIView):
+    def get(self, request):
+        genres = Genre.objects.all()
+        genre_serializer = GenreSerializer(genres, many=True)
+        return Response(genre_serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "POST":
-        serializer = MovieSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
+    def post(self, request):
+        genre_serializer = GenreSerializer(data=request.data)
+        genre_serializer.is_valid(raise_exception=True)
+        genre_serializer.save()
+        return Response(genre_serializer.data, status=status.HTTP_201_CREATED)
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+class GenreDetail(APIView):
+    def get_object(self, pk):
+        return get_object_or_404(Genre, pk=pk)
 
-@api_view(["GET", "PUT", "DELETE"])
-def movie_detail(request, pk):
-    movie = get_object_or_404(Movie, pk=pk)
+    def get(self, request, pk):
+        genre = self.get_object(pk)
+        genre_serializer = GenreSerializer(genre)
+        return Response(genre_serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "GET":
-        serializer = MovieSerializer(movie)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+    def put(self, request, pk):
+        genre = self.get_object(pk)
+        genre_serializer = GenreSerializer(genre, data=request.data)
+        genre_serializer.is_valid(raise_exception=True)
+        genre_serializer.save()
+        return Response(genre_serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "PUT":
-        serializer = MovieSerializer(movie, data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
+    def delete(self, request, pk):
+        genre = self.get_object(pk)
+        genre.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-    if request.method == "DELETE":
-        movie.delete()
-        return Response(status=status.HTTP_204_NO_CONTENT)
+class ActorList(
+    generics.GenericAPIView,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin
+):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        return self.create(request, *args, **kwargs)
+
```

Comment:
There is no need to define methods that are already implemented in mixins. You could remove them. Besides that, place mixins in the fist place in inheritance.